profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSnapshotResultBuffer.h

```diff
@@ -179,27 +174,30 @@ inline void StackSnapshotResultBuffer::DetermineAppDomain(ThreadID threadId, ICo
 
 inline bool StackSnapshotResultBuffer::AddFrame(std::uintptr_t ip)
 {
-    if (_currentFramesCount >= MaxSnapshotStackDepth_Limit)
-    {
-        return false;
-    }
-
-    if (_currentFramesCount == MaxSnapshotStackDepth_Limit - 1)
-    {
-        _instructionPointers[_currentFramesCount++] = 0;
-        return false;
-    }
-
-    _instructionPointers[_currentFramesCount++] = ip;
-    return true;
+    return _callstack.Add(ip);
 }
 
 inline bool StackSnapshotResultBuffer::AddFakeFrame()
 {
     return AddFrame(0);
 }
 
-inline uintptr_t* StackSnapshotResultBuffer::Data()
+inline shared::span<uintptr_t> StackSnapshotResultBuffer::Data()
+{
+    return _callstack.Data();
+}
+
+inline Callstack StackSnapshotResultBuffer::GetCallstack()
 {
-    return _instructionPointers.data();
+    return std::exchange(_callstack, {});
+}
+
+#include "Log.h"
+inline void StackSnapshotResultBuffer::SetCallstack(Callstack callstack)
+{
+    if (callstack.Capacity() == 0)
+    {
+        Log::Error("============ no space available");
```

Comment:
Maybe we will need to limit the number of log lines... Adding a metric could be helpful too

Reply:
oops debug ;) my bad